.vscode/launch.json

```diff
@@ -2,35 +2,40 @@
   "version": "0.2.0",
   "configurations": [
     {
-      "name": "C/C++: Linux: build and debug FreeCAD",
+      "name": "C/C++: build and debug FreeCAD",
       "type": "cppdbg",
       "request": "launch",
-      "program": "${workspaceFolder}/build/bin/FreeCAD",
+      "program": "${command:cmake.getLaunchTargetPath}",
       "args": [],
-      "stopAtEntry": false,
+      "stopAtEntry": true,
       "cwd": "${workspaceFolder}",
-      "environment": [],
-      "externalConsole": false,
-      "preLaunchTask": "CMake: build"
-    },
-    {
-      "name": "C/C++: macOS: build and debug FreeCAD",
-      "type": "cppdbg",
-      "request": "launch",
-      "program": "${workspaceFolder}/build/bin/FreeCAD",
-      "args": [],
-      "stopAtEntry": false,
-      "cwd": "${workspaceFolder}",
-      "environment": [],
+      "environment": [
+        {
+          "name": "PATH",
+          "value": "${command:cmake.getLaunchTargetDirectory}:${env:PATH}"
+        },
+        {
+          "name": "PYDEVD_DISABLE_FILE_VALIDATION",
+          "value": "1"
+        }
+      ],
+      "linux": {
+        "MIMode": "gdb",
+      },
+      "osx": {
+        "MIMode": "lldb"
+      },
+      "windows": {
+        "MIMode": "gdb",
+        "miDebuggerPath": "C:\\msys64\\ucrt64\\bin\\gdb.exe"
```

Comment:
Is this hardcoded absolute path necessary? What if someone has installed msys in another directory? Maybe msys provides some kind of env variable that we can use instead?

Reply:
I am not familiar with windows debugging, what would be the correct path? In my PR i have C:\\MinGw\\bin\\gdb.exe  and I cannot recall where its coming from. Also I've seen tasks of "type": "vscppdbg" instead of "cppdbg". Whats the correct way?

Reply:
That depends on the setup of the user. Most of windows users probably depends on msvc toolkit (the one from visual studio) which uses "vscppdbg" as the type. Others use gcc + gdb (coming from mingw/msys2) which must be installed separately and is not so common. I don't think that we should force anything like that. 

As for the correct path - there is no correct absolute path as it can be customized by users. I think that most msys users have the tools added to their PATH env variable, so simple `"gdb.exe"` should be good enough, we can always add that if users want to use gdb it should be available in PATH.

Reply:
Thanks, fixed

Reply:
does this need to be specified? can't we leave this blank and let users set it on their user config?

Reply:
I derived this by following the steps provided by the [VSCode Debugging C++ guide](https://code.visualstudio.com/docs/cpp/cpp-debug) (note: they have a different path than what their instructions result in).  While I had this line included, I never successfully got the debugger to attach to the running FreeCAD after launching it.

However @kadet1090 noted that when using the MSVC++ tools, which the official FreeCAD on Windows does (MSVC++ 2019), the appropriate debugger is `cppvsdbg` as seen on the [Microsoft C++ on Windows](https://code.visualstudio.com/docs/cpp/config-msvc#_customize-debugging-with-launchjson) guide.

I am currently performing a build on Windows and will test this configuration.  If functional, I believe that `cppvsdbg` should be the default with the option for user to override as needed.

Reply:
First attempt failed on my Macbook Pro running Apple M2.  It appears there's an issue when debuggin on Win 11 for ARM64 for a Windows x86/64 binary. [vscode-cpptools#10596](https://github.com/microsoft/vscode-cpptools/issues/10596)

I have a 10 year old Windows laptop that I'll attempt to build and test on.

Reply:
I did manage to build and debug in Windows 10 on x86-64, however the launch from debugger consistently failed wheras attaching to a running instance worked as expected.

I noticed that the behavior on Windows when running from the command line differs from Linux and macOS. Under Linux and macOS, launching from the command line the process is attached to the terminal.  Under Windows, the program forks and detaches, returning the user to a command prompt.  I am uncertain if this behavioral difference is responsible for the debugger failing to launch the program without error.

Reply:
Alright looks like we need an entirely new debug config for windows. I'll try to setup a windows dev env when I have some time next week. So we need `cppvsdbg` and `attach` yeah?

Reply:
From what I can tell, this is the best option for Windows development in VS Code.  I suspect most development is on Linux and macOS, but debugging Windows-specific issues will require Windows tooling.

Reply:
[This commit](https://github.com/FreeCAD/FreeCAD/pull/11172/commits/b43a19e7dad1da39e8058a04ce2dd123375c6656) shows how to add the debugger attach for Windows in `launch.json`.  Unfortunately, I have not seen a way to have launch configurations that are platform-specific.  According to the [documentation](https://code.visualstudio.com/Docs/editor/debugging#_platformspecific-properties) it seems that only properties can be platform-specific.